app/src/pages/Play/index.tsx

```diff
@@ -45,69 +41,30 @@ function Play() {
     (player) => player.id === activeTurn?.player_id
   )
 
-  const [activeTurnPlayState, setActiveTurnPlayState] = React.useState(
-    ActiveTurnPlayState.Waiting
-  )
-
-  const startingSeconds =
-    activeTurn?.seconds_per_turn_override || currentGame.seconds_per_turn || 0
-
-  const [secondsLeft, setSecondsLeft] = React.useState(
-    activeTurn?.started_at
-      ? calculateSecondsLeft(
-          dateFromTimestamptzNow(activeTurn.started_at),
-          startingSeconds
-        )
-      : startingSeconds
-  )
-
-  // change in settings
-  React.useEffect(() => {
-    if (currentGame.seconds_per_turn) {
-      setSecondsLeft(currentGame.seconds_per_turn)
+  const getActiveTurnPlayState = () => {
+    if (activeTurn?.review_started_at) {
+      return ActiveTurnPlayState.Reviewing
     }
-  }, [currentGame.seconds_per_turn])
 
-  // for non active players
-  React.useEffect(() => {
-    if (activeTurn?.started_at !== null) {
-      setActiveTurnPlayState(ActiveTurnPlayState.Playing)
-      setSecondsLeft(
-        calculateSecondsLeft(
-          dateFromTimestamptzNow(activeTurn?.started_at),
-          startingSeconds
-        )
-      )
+    if (activeTurn?.started_at) {
+      return ActiveTurnPlayState.Playing
     }
-  }, [activeTurn?.started_at])
 
-  // new turn, reset state
+    return ActiveTurnPlayState.Waiting
+  }
+  const [activeTurnPlayState, setActiveTurnPlayState] = React.useState(
+    getActiveTurnPlayState()
+  )
   React.useEffect(() => {
-    setActiveTurnPlayState(ActiveTurnPlayState.Waiting)
-    setSecondsLeft(
-      activeTurn?.started_at
-        ? calculateSecondsLeft(
-            dateFromTimestamptzNow(activeTurn.started_at),
-            startingSeconds
-          )
-        : startingSeconds
-    )
-  }, [activeTurn?.id])
+    setActiveTurnPlayState(getActiveTurnPlayState())
+  }, [activeTurn]) // eslint-disable-line react-hooks/exhaustive-deps
+
+  const serverTimeOffset = useServerTimeOffset()
```

Comment:
Open to suggestions on when – and how often we do this synchronization. It currently happens once when the play state renders.

Reply:
Yeah, I like that idea! Let me know if you're leaning one way or the other with regards where exactly to put it (context, local storage, etc) – I'm easy.

I think the synchronization should ideally happen:

1. at a time in which there are minimal other network requests
2. ~close to when it's actually used (i.e. active play time)

but it might not actually matter in practice. I'll think on it a little more, but maybe we at least wait until the user joins a game?

Reply:
7ad7c51a2add9b3c2082e2036d4c4ab21376e74d moves the synchronization into `CurrentPlayerProvider`, and stores it in the existing `CurrentPlayerContext` but still open to better options!